camel/agents/knowledge_graph_agent.py

```diff
@@ -0,0 +1,223 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from typing import Any, Optional, Union
+
+from unstructured.documents.elements import Element
+
+from camel.agents import ChatAgent
+from camel.messages import BaseMessage
+from camel.prompts import TextPrompt
+from camel.storages.graph_storages.graph_element import (
+    GraphElement,
+    Node,
+    Relationship,
+)
+from camel.types import ModelType, RoleType
+
+text_prompt = """
+You are tasked with extracting nodes and relationships from given content and structures them into Node and Relationship objects. Here's the outline of what you needs to do:
+
+Content Extraction:
+You should be able to process input content and identify entities mentioned within it.
+Entities can be any noun phrases or concepts that represent distinct entities in the context of the given content.
+
+Node Extraction:
+For each identified entity, you should create a Node object.
+Each Node object should have a unique identifier (id) and a type (type).
+Additional properties associated with the node can also be extracted and stored.
+
+Relationship Extraction:
+You should identify relationships between entities mentioned in the content.
+For each relationship, create a Relationship object.
+A Relationship object should have a subject (subj) and an object (obj) which are Node objects representing the entities involved in the relationship.
+Each relationship should also have a type (type), and additional properties if applicable.
+
+Output Formatting:
+The extracted nodes and relationships should be formatted as instances of the provided Node and Relationship classes.
+Ensure that the extracted data adheres to the structure defined by the classes.
+Output the structured data in a format that can be easily validated against the provided code.
+
+Instructions for you:
+Read the provided content thoroughly.
+Identify distinct entities mentioned in the content and categorize them as nodes.
+Determine relationships between these entities and represent them as directed relationships.
+Provide the extracted nodes and relationships in the specified format below.
+Example for you:
+
+Example Content:
+"John works at XYZ Corporation. He is a software engineer. The company is located in New York City."
+
+Expected Output:
+
+Nodes:
```

Comment:
For KG we usually have Entity and Relation. Maybe we can change node to entity and relationship to Relation.

Reply:
In Neo4j they use `Node` and `Relationship` as the screenshot below, how about keep the naming align with Neo4j?

<img width="245" alt="Screenshot 2024-05-23 at 01 15 11" src="https://github.com/camel-ai/camel/assets/133094783/5b193d36-29e0-4622-a7d8-646a2d18ccf2">